data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java

```diff
@@ -122,28 +126,54 @@ public void processUpgrade(String s, String s1) {
   }
 
   private void updateNewsImagesPermissions(Node newsNode, Session session) throws RepositoryException {
-    Matcher matcher = IMAGE_SRC_PATTERN.matcher(getStringProperty(newsNode, "exo:body"));
+    Matcher matcher = Pattern.compile(IMAGE_SRC_REGEX).matcher(getStringProperty(newsNode, "exo:body"));
     int imagesCount = 0;
-    while (matcher.find()) {
-      String match = matcher.group(1);
-      String imageUUID = match.substring(match.lastIndexOf("/") + 1);
-      ExtendedNode image = (ExtendedNode) session.getNodeByUUID(imageUUID);
-      if (image != null) {
-        if (image.canAddMixin(EXO_PRIVILEGEABLE)) {
-          image.addMixin(EXO_PRIVILEGEABLE);
+    ExtendedNode image = null;
+    if (matcher.find()) {
+      do {
+        String match = matcher.group(1);
+        String imageUUID = match.substring(match.lastIndexOf("/") + 1);
+        image = (ExtendedNode) session.getNodeByUUID(imageUUID);
+        if (image != null) {
+          if (image.canAddMixin(EXO_PRIVILEGEABLE)) {
+            image.addMixin(EXO_PRIVILEGEABLE);
+          }
+          boolean isPublicImage = image.getACL()
+                                       .getPermissionEntries()
+                                       .stream()
+                                       .anyMatch(accessControlEntry -> accessControlEntry.getIdentity()
+                                                                                         .equals(PLATFORM_USERS_GROUP_IDENTITY));
+          if (!isPublicImage) {
+            // make news images public
+            image.setPermission(PLATFORM_USERS_GROUP_IDENTITY, READ_PERMISSIONS);
+            image.save();
+            imagesCount += 1;
+          }
         }
-        boolean isPublicImage = image.getACL()
-                                     .getPermissionEntries()
-                                     .stream()
-                                     .filter(accessControlEntry -> accessControlEntry.getIdentity()
-                                                                                     .equals(PLATFORM_USERS_GROUP_IDENTITY))
-                                     .toList()
-                                     .size() > 0;
-        if (!isPublicImage) {
-          // make news images public
-          image.setPermission(PLATFORM_USERS_GROUP_IDENTITY, READ_PERMISSIONS);
-          image.save();
-          imagesCount += 1;
+      } while (matcher.find());
+    } else {
```

Comment:
What if we have both kind of image URLs inside the same body ? shouldn't we check both regex ?

Reply:
done!